### PR TITLE
doc: Reword docker integration text

### PR DESCRIPTION
**Pull request by raybejjani**
Some simple rewording. I skimmed the docker compose guide and I'm not sure if that's in scope here or not. I didn't change it.

### CODE REVIEW

**Comment by raybejjani** [comment]
test-me-please

File: Documentation/docker/index.rst [hunk 1]
```diff
@@ -4,28 +4,27 @@ Docker
...
-delegates control over IP address management and network connectivity to Cilium
-for all containers attached to this network for both IPv4 and IPv6
-connectivity.  Each Docker container gets an IP address from the node prefix of
-the node running the container.
-
-When deployed with Docker, each Linux node runs a ``cilium-docker`` agent,
-which receives libnetwork calls from Docker and then communicates with the
+* via the `CNI` interface. This method is used by `Kubernetes` and :ref:`Mesos`.
+* via Docker's `libnetwork`_ plugin interface, if networking is to be managed by
+  the Docker runtime. This method is used, for example, by Docker compose.
```
Comment (ianvernon) [inline Documentation/docker/index.rst]: Docker compose --> Docker Compose . Maybe link to Docker Compose?

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: Documentation/docker/index.rst (hunk 1 above)
Comment (raybejjani) [reply Documentation/docker/index.rst]: I linked to the docs but I can link to the github repo if you prefer. I also fixed the capitalisation in the docker compose gsg.

**Comment by raybejjani** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**